src/extension/wps-download/src/main/java/org/geoserver/wps/gs/download/DownloadMapProcess.java

```diff
@@ -301,6 +305,9 @@ RenderedImage buildImage(
         if (time != null) {
             template.put("time", time);
         }
+        if (animateParam != null && animateValue != null) {
```

Comment:
See what' happening here? You're basically doing the same as time, but with generic parameters.
I'd suggest to get rid of the time parameter altoghether, and fill the param name as "time" in the caller (that is, use the generic parameter path at this point). The method interface is broken anyways, might as well keep it clean.